apps/labs/posts/numba-polynomial-support.md

```diff
@@ -0,0 +1,173 @@
+fairly simple as it has three parameters - `coef`, `domain`, and `window`, each
+of which is a NumPy array. This meant that the foundation of the implementation
+was fairly straightforward. Unfortunately, there were a lot of details in the
+behaviour of the class, and each small detail seemed challenging when I had to
+implement it using the low-level API. The most interesting thing about the
+implementation of the class is that the two optional parameters - `domain` and
+`window`, have default values that are NumPy arrays of integer dtype, whereas if
+the user decides to set `domain` and `window`, then these arrays are always cast
+to arrays of double dtype. This meant that the user could never set the values
+of these two parameters to their default values. I have opened
+[pull request #24745](https://github.com/numpy/numpy/pull/24745) in NumPy that
+aims to improve this behaviour for all classes in the `numpy.polynomial`
+package.
+
+## Adding support for Power Series in Numba
+
+Similarly to functions, NumPy classes have to be reimplemented in Numba using
+its low-level API. A good example of this can be found in the
+[Numba documentation](https://numba.readthedocs.io/en/stable/extending/interval-example.html#example-an-interval-type),
+which I followed very closely. The first big task when working on the
+`Polynomial` class was defining a new Numba type to represent the class. After
+creating a type class called `PolynomialType`, I needed to "show" Numba that
+each instance of the `Polynomial` class should be treated by Numba as if its
+type is `PolynomialType`, using the
+`@typeof_impl.register(np.polynomial.polynomial.Polynomial)` decorator. Another
+important part is defining the constructor for the Polynomial class. We decided
+to add support for `Polynomial(coef)` and `Polynomial(coef, domain, window)` at
+this point. The next step is to define a data model for the new type. That is,
+how the Polynomial type will be represented in LLVM IR.
+
+The final task when adding support for a new type in Numba is defining special
+methods for converting Python objects into native values and vice versa. Those
+operations are called boxing and unboxing. The idea being that a Python object
+can be seen as a sophisticated box containing a simple native value which is
+what the data model needs. The tricky part in my implementation of those two was
+that I had to keep track of whether the `domain` and `window` were defined by
+the user or they equal the default value. This part is necessary since the
+default values of those two parameters can never be set using the constructor,
+as mentioned above.
+
+## Further work on support for NumPy polynomials
+
+Thanks to the support I have received from my mentor -
+[Guilherme Leobas](https://github.com/guilhermeleobas). After the respective
+pull request is closed, there is a lot of room for further work. For example,
+NumPy supports 5 more types of polynomials - Chebyshev Series, two types of
```

Comment:
Per styleguide, usually spell with words for 20 and below.

I'm _guessing_ that 'series' should be lowercase here, but I'm not sure.

```suggestion
NumPy supports five more types of polynomials: Chebyshev series, two types of
```